float32[] leftPoints #(x1,y1,x2,y2)
float32[] rightPoints #(x1,y1,x2,y2)
